File: src/proficiency/math/prime_checker/prime_checker.py
import sys
import math

def prime_checker(n):
    if n < 2:
        return False
    elif n == 2:
        return True
    elif n % 2 == 0:
        return False
    else:
        for i in range(3, int(math.sqrt(n)), 2):
            if n % i == 0:
                return False
        else:
            return True

print(prime_checker(int(sys.argv[1])))

File: src/proficiency/math/prime_checker/prime_checker.clj
(ns proficiency.math
  (:gen-class)
  (:require [clojure.math.numeric-tower :as math]))

(defn prime_checker [n]
  (cond
    (< n 2) false
    (= n 2) true
    (even? n) false
    (not-any? #(= (mod n %) 0) (range 3 (math/sqrt n) 2)) false
    :else true))

(defn -main
  [& args]
  (def n (Integer. (first args)))
  (println (prime_checker n)))
